interpreter/cling/lib/Interpreter/Value.cpp

```diff
@@ -258,6 +258,37 @@ namespace cling {
                                                     GetNumberOfElements());
   }
 
+  template <typename T> T convert(clang::QualType QT, Value::Storage& S) {
+    if (const auto *ET = llvm::dyn_cast<clang::EnumType>(QT.getTypePtr()))
+      QT = ET->getDecl()->getIntegerType();
+
+    QT = QT.getCanonicalType();
+
+    assert(QT->isBuiltinType());
+
+    const auto* BT = llvm::cast<const clang::BuiltinType>(QT.getTypePtr());
+    switch (BT->getKind()) {
+    default:
+      llvm_unreachable("Unknown type");
+#define X(type, name) \
+      case clang::BuiltinType::name: return (T) S.m_##name;
```

Comment:
Doesn't that lead to return the address of a temporary value in the `non-const` version of `getAs` ?

Reply:
Yes, I've added `setX` interfaces. Returning an address from the `get` is not very good because the user can cache the address and in the meanwhile we can destroy the `cling::Value` which the user cannot possibly know about. Now that's also solved except for one usecase called `getPtrAddress` which we need a little more work to remove.